### PR TITLE
Report sensor data using sensor API

### PR DESCRIPTION
**Pull request by stsoe**
#### Problem solved by the commit
Fix code to not go via xrt::device::get_info(), which is an end-user API, not a tool API.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Report function is called with a mgmtpf device, which was incorrectly converted to
userpf device because the device id was reopened as an xrt::device object.

#### How problem was solved, alternative solutions (if any) and why they were rejected
Core XRT (implies tools) should use xrt_core::device exclusively.   The core device is opened
explicitly as a mgmtpf device by xbmgmt thus all query calls operate on the mgmtpf device 
expected.   Also, it was wasteful to first convert ptree to string only to read the string back as a ptree, now the code stays with ptree.

#### Risks (if any) associated the changes in the commit
Can't think of any

#### What has been tested and how, request additional testing if necessary
Manual xbmgmt test before and after

### CODE REVIEW

File: src/runtime_src/core/tools/common/ReportMechanical.cpp [hunk 1]
```diff
@@ -18,35 +18,31 @@
 // Local - Include Files
 #include "ReportMechanical.h"
 
+#include "core/common/sensor.h"
+
```
Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportMechanical.cpp:22]: I don't think we need json_parser.hpp anymore

**Review by AShivangi — COMMENTED** [review]
Thanks for making this change! Should we make this change for all the reports?

File: src/runtime_src/core/tools/common/ReportMechanical.cpp (hunk 1 above)
Comment (stsoe) [reply src/runtime_src/core/tools/common/ReportMechanical.cpp:22]: I removed it.  Thanks!

**Comment by dayeh-xilinx** [comment]
retest this please. Jenkins server reboot.

**Comment by dayeh-xilinx** [comment]
retest this please. Due to Jenkins server reboot.

**Review by chvamshi-xilinx — APPROVED** [review]
This looks simpler . Are you making this change in all reports?

**Pull request merged**